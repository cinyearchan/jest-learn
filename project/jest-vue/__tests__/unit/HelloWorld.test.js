import { shallowMount } from '@vue/test-utils'
import HelloWorld from '@/components/HelloWorld.vue'
// import Vue from 'vue'

// describe('HelloWorld.vue', () => {
//   it('renders props.msg when passed', () => {
//     const msg = 'new message'
//     const wrapper = shallowMount(HelloWorld, {
//       propsData: { msg }
//     })
//     expect(wrapper.text()).toMatch(msg)
//   })
// })

// eslint-disable-next-line no-undef
describe('HelloWorld.vue', () => {
  /*
  // eslint-disable-next-line no-undef
  it('renders props.msg when passed', () => {
    const root = document.createElement('div')
    root.className = 'root'
    document.body.appendChild(root)
    new Vue({
      render: h => h(HelloWorld, {
        props: {
          msg: 'dell lee'
        }
      })
    }).$mount('.root')
    console.log(document.body.innerHTML)
    // eslint-disable-next-line no-undef
    expect(document.getElementsByClassName('hello').length).toBe(1)
  })
  */

  // eslint-disable-next-line no-undef
  it('renders props.msg when passed', () => {
    const msg = 'dell lee'
    const wrapper = shallowMount(HelloWorld, {
      propsData: { msg }
    })
    // eslint-disable-next-line no-undef
    expect(wrapper.findAll('.mmm').length).toBe(1)
    // eslint-disable-next-line no-undef
    expect(wrapper.props('msg')).toEqual(msg)
  })

  // eslint-disable-next-line no-undef
  it('组件渲染正常', () => {
    const wrapper = shallowMount(HelloWorld, {
      propsData: { msg: 'dell lee' }
    })
    // eslint-disable-next-line no-undef
    expect(wrapper).toMatchSnapshot() // 常用快照对 UI 进行测试
  })
})
