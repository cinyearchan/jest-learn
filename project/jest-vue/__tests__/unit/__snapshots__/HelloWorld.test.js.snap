// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue 组件渲染正常 1`] = `
<div class="hello">
  <h1 class="mmm abc">dell lee</h1>
</div>
`;
